### ISSUE TITLE
Operations without a name has "" (the empty string) as operationName

### ISSUE THREAD

**Issue by sorenbs** [apollographql/subscriptions-transport-ws#174]
The specification states that operationName on the GQL_START message is of type `operationName?: string`. 

Calling subscribeToMore with a unnamed subscription query causes the client to send `operationName: ""` (the empty string)

I think it would be preferable to let the operationName be `null` in this case.

---

Example subscription query:

```graphql
subscription {
    Location(filter: {
      mutation_in: [CREATED, UPDATED]
    }) {
        mutation
        node {
            id
            latitude
            longitude
            traveller {
                id
                name
            }
        }
    }
}
```

example subscription message:

```json
{"id":2,"type":"start","payload":{"query":"subscription {\n  Location(filter: {mutation_in: [CREATED, UPDATED]}) {\n    mutation\n    node {\n      id\n      latitude\n      longitude\n      traveller {\n        id\n        name\n        __typename\n      }\n      __typename\n    }\n    __typename\n  }\n}\n","variables":{},"operationName":""}}
```

**Comment by dotansimha** [comment]
@sorenbs 
This is not an issue with the transport or subscriptions.. this is an issue with `apollo-client` and the way is passes `operationName`.
When using `subscribeToMore`, Apollo client translates the document and extract the `operationName`, and it's default value is an empty string (see https://github.com/apollographql/apollo-client/blob/c3e14699cd945e8da8830e287111fa748e875428/src/queries/getFromAST.ts#L63-L69) 

The transport uses the same value provided by apollo-client, and does not change it at all.

**Comment by sorenbs** [comment]
Thanks for doing the detective work on this 👍

**Comment by dotansimha** [comment]
@sorenbs 
Just noticed that @helfer already released a version with a fix (1.4.2).
